pkg/api/api.go

```diff
@@ -73,13 +73,16 @@ type Config struct {
 	corsRegexString string `yaml:"cors_origin"`
 
 	buildInfoEnabled bool `yaml:"build_info_enabled"`
+
+	ProtobufQuerierHandler bool `yaml:"protobuf_querier_handler"`
 }
 
 // RegisterFlags adds the flags required to config this to the given FlagSet.
 func (cfg *Config) RegisterFlags(f *flag.FlagSet) {
 	f.BoolVar(&cfg.ResponseCompression, "api.response-compression-enabled", false, "Use GZIP compression for API responses. Some endpoints serve large YAML or JSON blobs which can benefit from compression.")
 	f.Var(&cfg.HTTPRequestHeadersToLog, "api.http-request-headers-to-log", "Which HTTP Request headers to add to logs")
 	f.BoolVar(&cfg.buildInfoEnabled, "api.build-info-enabled", false, "If enabled, build Info API will be served by query frontend or querier.")
+	f.BoolVar(&cfg.ProtobufQuerierHandler, "api.protobuf_querier_handler", false, "Enable protobuf serialization for range query and unsharded instant query responses from querier")
```

Comment:
I think we can remove this flag since we use codec now. I prefer to always install json and protobuf codec. Which codec to use is configured at client side (Query Frontend)

Reply:
A better way is probably always install all codecs. And change this flag to be `defaultCodec` so that users can pick whether to use JSON or Protobuf or something else.

Reply:
changed the config to be `QuerierDefaultCodec` with two possible string values `json` or `protobuf`, and `json` is the default. Do you think these should be different?